connectors/sources/aws.py

```
@@ -3,116 +3,223 @@
+        logger.debug(f"Downloading {filename}")
         async with self.session.client(
-            "s3", region_name=self.configuration["region"]
+            service_name="s3",
+            config=self.config,
+            endpoint_url=ENDPOINT_URL,
+            region_name=region,
         ) as s3:
-            # XXX limit the size
-            resp = await s3.get_object(Bucket=self.configuration["bucket"], Key=key)
-            await asyncio.sleep(0)
-            data = ""
-            while True:
-                chunk = await resp["Body"].read(ONE_MEGA)
+            try:
+                resp = await s3.get_object(Bucket=bucket, Key=filename)
                 await asyncio.sleep(0)
-                if not chunk:
-                    break
-                data += chunk.decode("utf8")
-            logger.debug(f"Downloaded {len(data)} for {key}")
-            return {"timestamp": timestamp, "text": data}
+                data = ""
+                while True:
+                    chunk = await resp["Body"].read(ONE_MEGA)
+                    await asyncio.sleep(0)
+                    if not chunk:
+                        break
+                    data += chunk.decode("utf8", errors="ignore")
+                logger.debug(f"Downloaded {len(data)} for {filename}")
+                return {"timestamp": timestamp, "text": data, "_id": doc["id"]}
+            except (ClientError, ServerTimeoutError, AioReadTimeoutError) as exception:
+                if (
+                    exception.response.get("Error", {}).get("Code")
+                    == "InvalidObjectState"
+                ):
+                    logger.error(
+                        f"{filename} of {bucket} is archived and inaccessible until restored. Error: {exception}"
+                    )
+                else:
+                    logger.error(
+                        f"Something went wrong while extracting data from {filename} of {bucket}. Error: {exception}"
+                    )
```

don't we want to reraise the exception here?

---

If we raise the exception then in case of glacier bucket type we will not be able to fetch the content and we will have to stop the execution and the further data will not get indexed. Is it ok?

---

Yes, we can raise the exception. However there is one edge case where if the user has glacier bucket type then the lib is not able to fetch the content and raises the exception and that will stop the execution if we reraise the exception. or we conditionally continue to iterate over next bucket in case the unsupported bucket storage type.

---

in case of Glacier you are right, we can warn the user and ignore the file